mpas_analysis/shared/interpolation/test_interpolate.py

```diff
@@ -0,0 +1,109 @@
+        
+        lat_input = np.deg2rad(np.array([0, 0, 0, 0, 45, 45, 45, 45]))
+        lon_input = np.deg2rad(np.array([0, 90, 180, 270, 0, 90, 180, 270]))
+        
+        lon_min = -90
+        lon_max = 90
+        lat_min = -45
+        lat_max = 45
+        dLon = 30
+        dLat = 30
+        
+        d, inds, lonTarg, latTarg = init_tree(lon_input, lat_input, lon_min, lon_max, 
+                                              lat_min, lat_max, dLon, dLat)
+                                              
+        nLonExpected = (lon_max - lon_min) / dLon
+        nLatExpected = (lat_max - lat_min) / dLat
+        
+        # LonTarg and LatTarg should be 2-D arrays
+        assert len(lonTarg.shape) == 2
+        assert len(latTarg.shape) == 2
+        
+        # LonTarg and latTarg should have expected sizes
+        assert lonTarg.shape[0] == nLonExpected
+        assert latTarg.shape[0] == nLonExpected
+        assert lonTarg.shape[1] == nLatExpected
+        assert latTarg.shape[1] == nLatExpected
+
+    def test_target_grid_bounds(self):
+        
+        lat_input = np.deg2rad(np.array([0, 0, 0, 0, 45, 45, 45, 45]))
+        lon_input = np.deg2rad(np.array([0, 90, 180, 270, 0, 90, 180, 270]))
+        
+        lon_min = -180
+        lon_max = 180
+        lat_min = -50
+        lat_max = 50
+        dLon = 20
+        dLat = 20
+        
+        d, inds, lonTarg, latTarg = init_tree(lon_input, lat_input, lon_min, lon_max, 
+                                              lat_min, lat_max, dLon, dLat)  
+                             
+        # LonTarg should respect bounds defined by user
+        assert lonTarg.max() <= lon_max
+        assert lonTarg.min() >= lon_min
+        
+        assert latTarg.max() <= lat_max
+        assert latTarg.min() >= lat_min
+        
+    def test_interp(self):
+        
+        lat_input = np.deg2rad(np.array([-45, -45, 45, 45]))
+        lon_input = np.deg2rad(np.array([-90, 90, -90, 90]))
+        vals_input = np.array([1, 2, 3, 4])
+        
+        lon_min = -90
+        lon_max = 90
+        lat_min = -45
+        lat_max = 45
+        dLon = 30
+        dLat = 30
+        
+        d, inds, lonTarg, latTarg = init_tree(lon_input, lat_input, lon_min, lon_max, 
+                                              lat_min, lat_max, dLon, dLat)  
+                                              
+        vals_output = interp_fields(vals_input, d, inds, lonTarg)
+        
+        #Test a few ramdom spots for nearest neighbor
+        #One exact spot
+        assert vals_output[0, 0] == vals_input[0]
+        
+        # Close point to val_input(0)
+        assert vals_output[2, 0] == vals_input[0]
+        
+        # Close to point val_input(1)
+        assert vals_output[4, 0] == vals_input[1]
+        
+        #Close to val_input(2)
+        assert vals_output[2, 2] == vals_input[2]
+        
+        #Close to val_input(3)
+        assert vals_output[4, 2] == vals_input[3]
+    
```

Comment:
There is some weird end of line character in line 109.  It looks like a circle crossed out and bent arrow in github.  I'm guessing you can just `dl` in vim to remove this issue at line 109.